core/src/main/scala/frameless/CatalystIntegral.scala

```diff
@@ -0,0 +1,20 @@
+package frameless
+
+import scala.annotation.implicitNotFound
+
+/** Types that can be remainded by Catalyst. */
+@implicitNotFound("Cannot do integral operation on columns of type ${A}.")
+trait CatalystIntegral[A]
+
+object CatalystIntegral {
```

Comment:
I assume that it's a good idea to have a separate typeclass for Integral types even if it represents the exact same set of types as numeric.

Reply:
I'm not super convinced, I think I would just merge them with a comment. We already reused some of these type functions if I recall correctly.